src/multi-selector.js

```diff
@@ -242,7 +249,8 @@ const MultiSelector = React.createClass({
 		// This is to allow adding of new items when the search term has matching filtered items but not an exact match
 		if (this.props.noRestrict && this.state.searchValue
 		&& !find(filterItems, (item) => getItemTitle(item).toLowerCase() === this.state.searchValue.toLowerCase())) {
-			filterItems.push(this.state.searchValue);
+			const searchValueToItem = this.props.searchValueToItem || this.searchValueToItem;
+			filterItems.push(searchValueToItem(this.state.searchValue));
```

Comment:
items is an array of unknown objects/strings/whatever else. So we provide a default behavior and allow for overriding the behavior if necessary